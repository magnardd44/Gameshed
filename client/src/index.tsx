import * as React from 'react';
import { Component } from 'react-simplified';
import ReactDOM from 'react-dom';
import {
  Card,
  Alert,
  Row,
  Form,
  Column,
  Button,
  NavBar,
  Header,
  Container,
  ColumnCentre,
  SearchResult,
} from './widgets';
import axios from 'axios';
import { HashRouter, Route } from 'react-router-dom';
import { AddReview, PublishReview } from './review-components';
import { Hash } from 'crypto';
import { createHashHistory } from 'history';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

const history = createHashHistory();

class NavHeader extends Component {
  render() {
    return (
      <div>
        <NavBar brand="GameShed">
          <NavBar.Link to="/games">Games</NavBar.Link>
        </NavBar>
        <Header></Header>
      </div>
    );
  }
}

class Search extends Component {
  input = '';
  render() {
    return (
      <>
        <Container textalign={'centre'}>
          <Row>
            <ColumnCentre width={6} offset={2}>
              <Form.Input
                type={this.input}
                value={this.input}
                placeholder="Søk etter et spill"
                onChange={(event) => (this.input = event.currentTarget.value)}
              />
            </ColumnCentre>
            <ColumnCentre width={2}>
              <Button.Success
                onClick={() => {
                  this.search();
                }}
              >
                Søk
              </Button.Success>
            </ColumnCentre>
          </Row>
        </Container>
      </>
    );
  }
  search() {
    console.log('search');
    history.push('/results');
  }
}

class SearchListings extends Component {
  render() {
    return (
      <Container>
        Søkeresultater:
        <SearchResult></SearchResult>
        <SearchResult></SearchResult>
        <SearchResult></SearchResult>
      </Container>
    );
  }
}

ReactDOM.render(
  <HashRouter>
    <div>
      <NavHeader />
      <Route exact path="/" component={Search}></Route>
      <Route exact path="/results" component={SearchListings}></Route>
      <Route exact path="/addReview" component={AddReview} />
      <Route exact path="/publishReview/:id" component={PublishReview} />
    </div>
  </HashRouter>,
  document.getElementById('root')
);

/*
class Start extends Component {
  input = '';
  stdout = '';
  stderr = '';
  errCode: number | null = null;
}



  render() {
    return (
      <>
        <Card title="GameShed">
          <Row>
            <Column>
              <Form.Label>Søk:</Form.Label>
              <Form.Textarea
                value={this.input}
                onChange={(event) => (this.input = event.currentTarget.value)}
              />
            </Column>
          </Row>
          <Button.Success
            onClick={() => {
              axios
                .post<{ exitStatus: number; stdout: string; stderr: string }>('/run', {
                  language: 'js',
                  source: this.input,
                })
                .then((response) => {
                  this.errCode = response.data.exitStatus;
                  this.stdout = response.data.stdout;
                  this.stderr = response.data.stderr;
                })
                .catch((error: Error) => Alert.danger('Could not run app.js: ' + error.message));
            }}
          >
            Søk etter spill
          </Button.Success>
        </Card>
        <Card title="Standard output">{this.stdout}</Card>
        <Card title="Standard error">{this.stderr}</Card>
        <Card title="Errorcode">{this.errCode}</Card>
      </>
    );
  }
}


const root = document.getElementById('root');
if (root)
  ReactDOM.render(
    <HashRouter>
      <Alert />

      <Route exact path="/" component={Start} />
      
    </HashRouter>,
    root
  );
 
 */
