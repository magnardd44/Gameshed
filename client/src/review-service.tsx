import axios from 'axios';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Review = {
  id: number;
  title: string;
  text: boolean;
  rating: number;
};

class ReviewService {
  /**
   * Get task with given id.
   */
  get(id: number) {
    return axios.get<Review>('/reviews/' + id).then((response) => response.data);
  }

  /**
   * Get all tasks.
   */
  getAll() {
    return axios.get<Review[]>('/reviews').then((response) => response.data);
  }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  create(title: string, text: string, rating: number) {
    return axios
      .post<{ id: number }>('/reviews', { review_title: title, text: text, rating: rating })
      .then((response) => response.data.id);
  }
}

const reviewService = new ReviewService();
export default reviewService;
