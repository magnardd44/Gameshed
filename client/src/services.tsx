import axios from 'axios';
import { sharedComponentData } from 'react-simplified';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Game = {
  game_id: number;
  game_title: string;
  genre: string[];
  genre_id: number;
  platform: string[];
  game_description: string;
};

export type Review = {
  review_id: number;
  game_id: string;
  external_game_id: string[];
  game_title: number;
  genre: string[];
  platform: string;
  review_title: string;
  text: string;
  user_id: number;
  rating: number;
  published: boolean;
};

class GameService {
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: [],
    genre_id: 0,
    platform: [],
    game_description: '',
  };
  games: Game[] = [];

  /**
   * Get game with given id.
   */
  get(id: number) {
    return axios.get<Game>('/games/' + id).then((response) => response.data);
  }

  /**
   * Get all games.
   */
  getAll() {
    return axios.get<Game[]>('/games').then((response) => response.data);
  }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  create(title: string, genre: string, platform: number, description: string) {
    return axios
      .post<{ id: number }>('/games', {
        game_title: title,
        genre: genre,
        platform: platform,
        game_description: description,
      })
      .then((response) => response.data.id);
  }
}

class ReviewService {
  review: Review = {
    review_id: 0,
    game_id: '',
    external_game_id: [],
    game_title: 0,
    genre: [],
    platform: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
  };
  reviews: Review[] = [];

  /**
   * Get review with given id.
   */
  get(id: number) {
    return axios.get<Review>('/reviews/' + id).then((response) => response.data);
  }

  /**
   * Get all reviews.
   */
  getAll() {
    return axios.get<Review[]>('/reviews').then((response) => response.data);
  }

  /**
   * Create new review having the given title.
   *
   * Resolves the newly created review id.
   */
  create(title: string, text: string, rating: number) {
    return axios
      .post<{ id: number }>('/reviews', { review_title: title, text: text, rating: rating })
      .then((response) => response.data.id);
  }
}

export let gameService = sharedComponentData(new GameService());
export let reviewService = sharedComponentData(new ReviewService());
