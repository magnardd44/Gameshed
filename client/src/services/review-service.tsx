import axios from 'axios';
import { sharedComponentData } from 'react-simplified';
import userService from './user-service';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Oauth_token = {
  access_token: string;
  expire_time: number;
  token_type: string;
};

class SearchService {
  token: Oauth_token = {
    access_token: '',
    expire_time: 0,
    token_type: '',
  };
}

export type Review = {
  review_id: number;
  game_id: number;
  game_title: string;
  review_title: string;
  text: string;
  user_id: number;
  rating: number;
  published: boolean;
  genre_id: number;
  platform_id: number;
  relevant: number;
  likes: number;
};

export type RelatedReview = {
  review_id: number;
  game_id: number;
  game_title: string;
  review_title: string;
  text: string;
  user_id: number;
  rating: number;
  published: boolean;
  genre_id: number;
  platform_id: number;
  relevant: number;
  likes: number;
};

class ReviewService {
  review: Review = {
    review_id: 0,
    game_id: 0,
    game_title: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
    genre_id: 0,
    platform_id: 0,
    relevant: 0,
    likes: 0,
  };
  reviews: Review[] = [];

  relatedReview: RelatedReview = {
    review_id: 0,
    game_id: 0,
    game_title: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
    genre_id: 0,
    platform_id: 0,
    relevant: 0,
    likes: 0,
  };

  relatedReviews: RelatedReview[] = [];

  publishedreviews: Review[] = [];

  drafts: Review[] = [];

  /**
   * Get review with given id.
   */
  get(id: number) {
    return userService.axios.get<Review>('/reviews/review/' + id).then((response) => response.data);
  }

  /**
   * Get review with given id.
   */
  getAllByGameId(id: number) {
    return userService.axios.get<Review>('/reviews/' + id).then((response) => response.data);
  }

  /**
   * Get reviews with given id.
   */
  getAllById(user_id: number) {
    return axios.get<Review[]>('/reviews/users/' + user_id).then((response) => response.data);
  }

  /**
   * Get reviews with given id.
   */
  getAll() {
    return axios.get<Review[]>('/reviews/').then((response) => response.data);
  }

  /**
   * Get published reviews
   */
  getPublishedReviews() {
    return axios.get<Review[]>('/reviews/published').then((response) => response.data);
  }

  /**
   * Get reviews based on genre
   */
  getGenre(genre_id: number) {
    return axios.get<Review[]>('/reviews/genre/' + genre_id).then((response) => response.data);
  }

  //Get reviews based on platform
  getPlatform(platform_id: number) {
    return axios
      .get<Review[]>('/reviews/platform/' + platform_id)
      .then((response) => response.data);
  }

  /**
   * Create new review   *
   * Resolves the newly created review id.
   */
  create(game_id: number, title: string, text: string, rating: number) {
    return userService.axios
      .post<{ id: number }>('/reviews', {
        game_id: game_id,
        review_title: title,
        text: text,
        rating: rating,
      })
      .then((response) => response.data.id);
  }

  //Change review status to published
  publish(id: number) {
    return userService.axios
      .patch<Review>('/reviews/' + id + '/publish')
      .then((response) => response.data.review_id);
  }
  //Add like to review
  like(review_id: number, user_id: number, relevant: number) {
    return userService.axios
      .patch<Review>('/reviews/' + review_id + '/relevant', {
        relevant: relevant,
        user_id: user_id,
      })
      .then((response) => response.data);
  }

  //Edit review before publishing
  edit(id: number, title: string, text: string, rating: number) {
    return userService.axios
      .patch<Review>('/reviews/' + id, { title: title, text: text, rating: rating })
      .then((response) => response.data);
  }

  //Delete review
  delete(id: number) {
    return userService.axios
      .delete<Review>('reviews/' + id)
      .then((response) => response.data.review_id);
  }
}

export let reviewService = sharedComponentData(new ReviewService());
