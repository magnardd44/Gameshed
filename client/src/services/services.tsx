import axios from 'axios';
import { sharedComponentData } from 'react-simplified';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Game = {
  game_id: number;
  game_title: string;
  genre: string[];
  genre_id: number;
  platform: string[];
  game_description: string;
};

class GameService {
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: [],
    genre_id: 0,
    platform: [],
    game_description: '',
  };
  games: Game[] = [];

  /**
   * Get game with given id.
   */
  get(id: number) {
    return axios.get<Game>('/games/' + id).then((response) => response.data);
  }

  /**
   * Get all games.
   */
  getAll() {
    return axios.get<Game[]>('/games').then((response) => response.data);
  }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  create(title: string, genre: string, platform: number, description: string) {
    return axios
      .post<{ id: number }>('/games', {
        game_title: title,
        genre: genre,
        platform: platform,
        game_description: description,
      })
      .then((response) => response.data.id);
  }
}

export let gameService = sharedComponentData(new GameService());
