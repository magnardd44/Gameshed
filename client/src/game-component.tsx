import * as React from 'react';
import { Component } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { gameService, reviewService, Game, Review } from './services';
import { createHashHistory } from 'history';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student

export class GameCard extends Component {
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: [],
    genre_id: 0,
    platform: [],
    game_description: '',
  };
  description =
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco...';
  render() {
    return (
      <Card title={this.game.game_title}>
        <h6 className="card-subtitle mb-2 text-muted">
          Terningkast:
          <ThumbNail small img="https://cdn-icons-png.flaticon.com/512/220/220725.png"></ThumbNail>
        </h6>
        <Row>
          <Column width={2}>
            <ThumbNail img="https://cdn-icons-png.flaticon.com/512/686/686589.png"></ThumbNail>
          </Column>
          <Column width={6}>
            {this.game.game_description}
            <Linebreak></Linebreak>
          </Column>
          <Column width={2}></Column>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <Column>Sjanger: {this.game.genre}</Column>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <Column>Platform: {this.game.platform}</Column>
        </Row>
      </Card>
    );
  }
  /** return axios
   */
  mounted() {
    gameService.getAll().then((result) => {
      this.game = result[0];
      console.log(this.game);
    });
  }
}
