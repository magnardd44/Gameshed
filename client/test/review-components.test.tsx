import * as React from 'react';
import { shallow } from 'enzyme';
import { useHistory } from 'react-router';
import { Card, Row, Column, Form, Button } from '../src/widgets';
import {
  AddReview,
  EditReview,
  PublishReview,
  CompleteReview,
  PublishedReviews,
  MyReviews,
} from '../src/review-components';
import { NavLink } from 'react-router-dom';

import { reviewService } from '../src/services/review-service';
import { FacebookIcon, FacebookShareButton } from 'react-share';
import { userInfo } from 'os';
import userService from '../src/services/user-service';
import MockAdapter from 'axios-mock-adapter/types';

jest.mock('../src/services/game-service', () => {
  class GameService {
    current = {
      game_id: 1,
      igdb_id: 0,
      game_title: '',
      genre: [],
      platform: [],
      game_description: '',
      igdb: null,
    };

    set = jest.fn().mockResolvedValue(this.current);
    get = jest.fn().mockResolvedValue(this.current);
  }
  return { gameService: new GameService() };
});

jest.mock('../src/services/review-service', () => {
  class ReviewService {
    reviews = [];

    review = { game_id: 1, review_id: 1, review_title: 'Elsker dette spillet' };

    getPublishedReviews() {
      return Promise.resolve([
        { game_id: 1, review_id: 1, review_title: 'test1', text: 'dette er en test1', rating: 4 },
        { game_id: 1, review_id: 2, review_title: 'test2', text: 'dette er en test2', rating: 2 },
        { game_id: 1, review_id: 3, review_title: 'test3', text: 'dette er en test2', rating: 3 },
      ]);
    }
    get(review_id: number) {
      return Promise.resolve({
        review_id: review_id,
        game_id: 1,
      });
    }
    getDraft(review_id: number) {
      return Promise.resolve({
        review_id: review_id,
        game_id: 1,
      });
    }

    getComplete(review_id: number) {
      return Promise.resolve({
        review_id: review_id,
        game_id: 1,
      });
    }

    // create() {
    //   return Promise.resolve(); // Same as: return new Promise((resolve) => resolve(4));
    // }

    create = jest.fn().mockResolvedValue(3);

    delete = jest.fn().mockResolvedValue(3);

    publish = jest.fn().mockResolvedValue(3);

    edit = jest.fn().mockResolvedValue(3);

    like = jest.fn().mockResolvedValue(3);

    getGenre = jest.fn().mockResolvedValue(3);

    getPlatform = jest.fn().mockResolvedValue(3);
  }
  return { reviewService: new ReviewService() };
});

describe('Review components tests', () => {
  // test('1 PublishedReviews draws correctly', (done) => {
  //   const wrapper = shallow(<PublishedReviews />);

  //   // Wait for events to complete
  //   setTimeout(() => {
  //     expect(
  //       wrapper.containsMatchingElement([
  //         <NavLink to="/publishedReviews/1">test1</NavLink>,
  //         <NavLink to="/publishedReviews/2">test2</NavLink>,
  //         <NavLink to="/publishedReviews/3">test3</NavLink>,
  //       ])
  //     ).toEqual(true);
  //     done();
  //   });
  // });

  test('save button in AddReview calls create method when clicked', (done) => {
    //@ts-ignore
    let newReview = new AddReview();
    //@ts-ignore
    const wrapper = shallow(<AddReview match={{ params: { review_id: 1 } }} />);
    wrapper.find(Button.Success).simulate('click');
    setTimeout(() => {
      //@ts-ignore
      if (
        reviewService.review.review_title !== '' ||
        reviewService.review.text !== '' ||
        reviewService.review.rating !== 0
      ) {
        expect(reviewService.create).toHaveBeenCalled();
      }
      done();
    });
  });

  test('delete button in PublishReview calls delete method when clicked', (done) => {
    //@ts-ignore
    let newReview = new PublishReview();
    //@ts-ignore
    const wrapper = shallow(<PublishReview match={{ params: { review_id: 1 } }} />);
    wrapper.find(Button.Danger).simulate('click');
    setTimeout(() => {
      //@ts-ignore
      expect(reviewService.delete).toHaveBeenCalled();

      done();
    });
  });

  test('publish button in PublishReview calls publish method when clicked', (done) => {
    //@ts-ignore
    let newReview = new PublishReview();
    //@ts-ignore
    const wrapper = shallow(<PublishReview match={{ params: { review_id: 1 } }} />);
    wrapper.find(Button.Success).at(1).simulate('click');
    setTimeout(() => {
      //@ts-ignore
      expect(reviewService.publish).toHaveBeenCalled();

      done();
    });
  });

  test('edit button in PublishReview calls history.push method when clicked', (done) => {
    //@ts-ignore
    const spy = jest.spyOn(history, 'push');

    //@ts-ignore
    let newReview = new PublishReview();
    //@ts-ignore
    const wrapper = shallow(<PublishReview match={{ params: { review_id: 1 } }} />);
    wrapper.find(Button.Success).at(0).simulate('click');
    setTimeout(() => {
      //@ts-ignore
      expect(history.push).toHaveBeenCalled();
      spy.mockRestore();

      done();
    });
  });

  test('save button in EditReview calls edit method when clicked', (done) => {
    //@ts-ignore
    let edit = new EditReview();
    //@ts-ignore
    const wrapper = shallow(<EditReview match={{ params: { review_id: 1 } }} />);
    wrapper.find(Button.Success).simulate('click');
    setTimeout(() => {
      //@ts-ignore
      expect(reviewService.edit).toHaveBeenCalled();

      done();
    });
  });

  test('like button in CompleteReview calls like method when clicked', (done) => {
    //@ts-ignore
    //let newReview = new CompleteReview();
    //@ts-ignore
    const wrapper = shallow(<CompleteReview match={{ params: { review_id: 1 } }} />);
    //wrapper.find(Button.Success).simulate('click');
    setTimeout(() => {
      //@ts-ignore
      //expect(reviewService.like).toHaveBeenCalled();

      done();
    });
  });

  test('publish button in MyReviews calls publish method when clicked', (done) => {
    //@ts-ignore
    let newReview = new MyReviews();
    let sikker = confirm('Er du sikker på at du vil publisere denne anmeldelsen?');
    //@ts-ignore
    const wrapper = shallow(<MyReviews match={{ params: { review_id: 1 } }} />);
    if (userService.token && reviewService.reviews.length > 0) {
      wrapper.find(Button.Success).at(1).simulate('click');
    }
    setTimeout(() => {
      //@ts-ignore
      if (sikker) {
        expect(reviewService.publish).toHaveBeenCalled();
      }

      done();
    });
  });

  test('delete button in MyReviews calls delete method when clicked', (done) => {
    //@ts-ignore
    let newReview = new MyReviews();
    let sikker = confirm('Er du sikker på at du vil slette denne anmeldelsen?');
    //@ts-ignore
    const wrapper = shallow(<MyReviews match={{ params: { review_id: 1 } }} />);
    if (userService.token && reviewService.reviews.length > 0) {
      wrapper.find(Button.Danger).at(0).simulate('click');
    }
    setTimeout(() => {
      //@ts-ignore
      if (sikker) {
        expect(reviewService.delete).toHaveBeenCalled();
      }

      done();
    });
  });

  test('Complete review draws FB share button', (done) => {
    const shareButtonProps = {
      url: 'https://localhost:3000/#/publishedReviews/1',
      network: 'Facebook',
      text: 'Tror du vil like denne anmeldelsen',
      longtext:
        'Social sharing buttons for React. Use one of the build-in themes or create a custom one from the scratch.',
    };
    //@ts-ignore
    const wrapper = shallow(<CompleteReview match={{ params: { review_id: 1 } }} />);
    console.log(wrapper.debug());

    // Wait for events to complete
    setTimeout(() => {
      expect(
        wrapper
          .find(
            'ForwardRef(ShareButton-facebook)'
            // <FacebookShareButton {...shareButtonProps}>
            //   <FacebookIcon size="40" round />
            // </FacebookShareButton>
          )
          .exists()
      ).toEqual(true);
      done();
    });
  });

  test('Complete review draws email share button', (done) => {
    const shareButtonProps = {
      url: 'https://localhost:3000/#/publishedReviews/1',
      network: 'Facebook',
      text: 'Tror du vil like denne anmeldelsen',
      longtext:
        'Social sharing buttons for React. Use one of the build-in themes or create a custom one from the scratch.',
    };
    //@ts-ignore
    const wrapper = shallow(<CompleteReview match={{ params: { review_id: 1 } }} />);
    console.log(wrapper.debug());

    // Wait for events to complete
    setTimeout(() => {
      expect(
        wrapper
          .find(
            'ForwardRef(ShareButton-email)'
            // <FacebookShareButton {...shareButtonProps}>
            //   <FacebookIcon size="40" round />
            // </FacebookShareButton>
          )
          .exists()
      ).toEqual(true);
      done();
    });
  });

  test.skip('x addReview correctly sets path on create', (done) => {
    //@ts-ignore
    const wrapper = shallow(<AddReview match={{ params: { db_id: 4 } }} />);

    wrapper
      .find(Form.Input)

      .simulate('change', { currentTarget: { value: 'Elsker dette spillet' } });

    // @ts-ignore
    expect(wrapper.containsMatchingElement(<Form.Input value="Elsker dette spillet" />)).toEqual(
      true
    );

    wrapper.find(Button.Success).simulate('click');
    // Wait for events to complete
    setTimeout(() => {
      expect(location.hash).toEqual('/publishedReviews/4');
      done();
    });
  });
});

describe('Snapshot component tests', () => {
  test('1 Snapshot - AddReview draws correctly', () => {
    const match = { params: { id: 1 } };
    //@ts-ignore
    const wrapper = shallow(<AddReview match={match} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('2 Snapshot - EditReview draws correctly', () => {
    const match = { params: { id: 1 } };
    //@ts-ignore
    const wrapper = shallow(<EditReview match={match} />);

    expect(wrapper).toMatchSnapshot();
  });
  test('3 Snapshot - CompleteReview draws correctly', () => {
    const match = { params: { id: 1 } };
    //@ts-ignore
    const wrapper = shallow(<CompleteReview match={match} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('4 Snapshot - PublishReview draws correctly', () => {
    const match = { params: { id: 1 } };
    //@ts-ignore
    const wrapper = shallow(<PublishReview match={match} />);

    expect(wrapper).toMatchSnapshot();
  });
  test('5 Snapshot - MyReviews draws correctly', () => {
    const match = { params: { id: 1 } };

    const wrapper = shallow(<MyReviews match={match} />);

    expect(wrapper).toMatchSnapshot();
  });
});
