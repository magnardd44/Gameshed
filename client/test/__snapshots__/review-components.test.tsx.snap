// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot component tests 1 Snapshot - AddReview draws correctly 1`] = `
<Container>
  <Card
    title="Skriv anmeldelse"
  >
    <Row>
      <Column
        width={2}
      >
        Spill:
      </Column>
      <Column />
    </Row>
    <Row>
      <Column
        width={2}
      >
        Sjanger:
      </Column>
      <Column />
    </Row>
    <Row>
      <Column
        width={2}
      >
        Plattform:
      </Column>
      <Column />
    </Row>
    <FormContainer>
      <FormGroup>
        <FormLabel>
          Overskrift:
        </FormLabel>
        <FormInput
          onChange={[Function]}
          placeholder="Skriv inn overskriften her:"
          type="text"
        />
      </FormGroup>
      <FormGroup>
        <FormLabel>
          Anmeldelse:
        </FormLabel>
        <FormTextarea
          cols={10}
          onChange={[Function]}
          placeholder="Skriv inn anmeldelsen her:"
          rows={10}
          value=""
        />
      </FormGroup>
      <FormGroup>
        <FormLabel>
          Terningkast:
        </FormLabel>
        <FormSelect
          onChange={[Function]}
          placeholder={
            <div>
              Velg terningkast:
            </div>
          }
        >
          <option
            key="0"
            value={1}
          >
            1
          </option>
          <option
            key="1"
            value={2}
          >
            2
          </option>
          <option
            key="2"
            value={3}
          >
            3
          </option>
          <option
            key="3"
            value={4}
          >
            4
          </option>
          <option
            key="4"
            value={5}
          >
            5
          </option>
          <option
            key="5"
            value={6}
          >
            6
          </option>
        </FormSelect>
      </FormGroup>
    </FormContainer>
  </Card>
  <Linebreak />
  <Column>
    <ButtonDanger
      onClick={[Function]}
    >
      Tilbake
    </ButtonDanger>
  </Column>
  <Column
    right={true}
  >
    <ButtonSuccess
      onClick={[Function]}
    >
      Lagre
    </ButtonSuccess>
  </Column>
</Container>
`;

exports[`Snapshot component tests 2 Snapshot - EditReview draws correctly 1`] = `
<Container>
  <Card
    title="Edit review"
  >
    <Row>
      <Column
        width={2}
      >
        Spill:
      </Column>
      <Column />
    </Row>
    <Row>
      <Column
        width={2}
      >
        Sjanger:
      </Column>
      <Column />
    </Row>
    <Row>
      <Column
        width={2}
      >
        Plattform:
      </Column>
      <Column />
    </Row>
    <Linebreak />
    <FormContainer>
      <FormGroup>
        <FormLabel>
          Title:
        </FormLabel>
        <FormInput
          onChange={[Function]}
          type="text"
          value=""
        />
      </FormGroup>
      <FormGroup>
        <FormLabel>
          Text:
        </FormLabel>
        <FormTextarea
          onChange={[Function]}
          rows={10}
          value=""
        />
      </FormGroup>
      <FormGroup>
        <FormLabel>
          Terningkast:
        </FormLabel>
        <FormSelect
          onChange={[Function]}
          placeholder={
            <div>
              Velg terningkast:
            </div>
          }
          value={0}
        >
          <option
            key="0"
            value={1}
          >
            1
          </option>
          <option
            key="1"
            value={2}
          >
            2
          </option>
          <option
            key="2"
            value={3}
          >
            3
          </option>
          <option
            key="3"
            value={4}
          >
            4
          </option>
          <option
            key="4"
            value={5}
          >
            5
          </option>
          <option
            key="5"
            value={6}
          >
            6
          </option>
        </FormSelect>
      </FormGroup>
    </FormContainer>
  </Card>
  <Row>
    <Column>
      <Linebreak />
      <ButtonSuccess
        onClick={[Function]}
      >
        Lagre
      </ButtonSuccess>
    </Column>
  </Row>
</Container>
`;

exports[`Snapshot component tests 3 Snapshot - CompleteReview draws correctly 1`] = `
<Container>
  <FullReviewCard
    img=" "
    subtitle=""
  >
    <Row>
      <Column
        width={2}
      >
        Sjanger: 
      </Column>
      <Column />
    </Row>
    <Row>
      <Column
        width={2}
      >
        Plattform:
      </Column>
      <Column />
    </Row>
    <Linebreak />
    <Row>
      <Column
        width={1}
      >
        <ButtonSuccess
          onClick={[Function]}
        >
          Like
        </ButtonSuccess>
      </Column>
      <Column
        width={1}
      >
        0
      </Column>
      <Column />
      <Column
        width={2}
      >
        <div>
           Del denne anmeldelsen:
        </div>
      </Column>
      <Column
        width={1}
      >
        <ForwardRef(ShareButton-facebook)
          longtext="Social sharing buttons for React. Use one of the build-in themes or create a custom one from the scratch."
          network="Facebook"
          text="Tror du vil like denne anmeldelsen"
          url="https://localhost:3000/#/publishedReviews/1"
        >
          <Icon
            bgStyle={Object {}}
            borderRadius={0}
            iconFillColor="white"
            round={true}
            size="40"
          />
        </ForwardRef(ShareButton-facebook)>
      </Column>
      <Column
        width={1}
      >
        <ForwardRef(ShareButton-email)
          longtext="Social sharing buttons for React. Use one of the build-in themes or create a custom one from the scratch."
          network="Facebook"
          subject="Tror du vil like denne spillanmeldelsen!"
          text="Tror du vil like denne anmeldelsen"
          url="https://localhost:3000/#/publishedReviews/1"
        >
          <Icon
            bgStyle={Object {}}
            borderRadius={0}
            iconFillColor="white"
            round={true}
            size="40"
          />
        </ForwardRef(ShareButton-email)>
      </Column>
    </Row>
  </FullReviewCard>
</Container>
`;

exports[`Snapshot component tests 4 Snapshot - PublishReview draws correctly 1`] = `
<Container>
  <Card
    title="Anmeldelse til publisering"
  >
    <Row>
      <Column
        width={2}
      >
        <b>
          Spill:
        </b>
      </Column>
      <Column />
    </Row>
    <Row>
      <Column
        width={2}
      >
        <b>
          Sjanger:
        </b>
      </Column>
      <Column />
    </Row>
    <Row>
      <Column
        width={2}
      >
        <b>
          Plattform:
        </b>
      </Column>
      <Column />
    </Row>
    <Linebreak />
    <Card
      title=""
    >
      <Row>
        <Column
          width={12}
        >
          <div>
            <b>
              Tittel: 
            </b>
          </div>
        </Column>
      </Row>
      <Linebreak />
      <Row>
        <Column
          width={12}
        >
          <div>
            <b>
              Innhold: 
            </b>
          </div>
        </Column>
      </Row>
      <Linebreak />
      <Row>
        <Column
          width={12}
        >
          <div>
            <b>
              Terningkast:
            </b>
             
          </div>
        </Column>
      </Row>
    </Card>
    <Linebreak />
    <Row>
      <Column>
        <ButtonSuccess
          onClick={[Function]}
        >
          Rediger
        </ButtonSuccess>
      </Column>
      <Column>
        <ButtonSuccess
          onClick={[Function]}
        >
          Publiser
        </ButtonSuccess>
      </Column>
      <Column>
        <ButtonDanger
          onClick={[Function]}
        >
          Slett
        </ButtonDanger>
      </Column>
      <Column>
        <ButtonLight
          onClick={[Function]}
        >
          Lagre som utkast
        </ButtonLight>
      </Column>
    </Row>
  </Card>
</Container>
`;

exports[`Snapshot component tests 5 Snapshot - MyReviews draws correctly 1`] = `
<Container>
  Du er ikke logget inn. Logg inn for Ã¥ se dine anmeldelser.
</Container>
`;
